org-roam-db.el

```diff
@@ -160,31 +138,24 @@ SQL can be either the emacsql vector representation, or a string."
   '((files
      [(file :unique :primary-key)
       (hash :not-null)
-      (meta :not-null)])
+      atime
+      mtime])
 
-    (ids
-     [(id :unique :primary-key)
+    (nodes
+     [(id :primary-key :not-null)
       (file :not-null)
-      (level :not-null)])
+      (level :not-null)
+      (pos :not-null)
```

Comment:
Is it position in the buffer? Do I understand it correctly that for `(= level 0)` the `pos` is `(point-min)` and for `(> level 0)`, it is the point at the beginning of heading? It is nice to have it baked. I search for it time to time.

I must say, I love this structure, this has even more than I usually use:

```emacs-lisp
(cl-defstruct vulpea-note
  id
  title
  path
  tags
  level)
```

And with this structure, it will be much easier to query database 💯 

Side note, do you consider using structs for passing around data queried from `db`?